R/getBaselineDivergence.R

```diff
@@ -38,168 +86,215 @@
 #' @importFrom SummarizedExperiment colData
 #' @importFrom SummarizedExperiment colData<-
 #' @importFrom SingleCellExperiment altExp
-#'
-#' @examples
-#' #library(miaTime)
-#' library(TreeSummarizedExperiment)
-#' library(dplyr)
-#'
-#' data(hitchip1006)
-#' tse <- mia::transformCounts(hitchip1006, method = "relabundance")
-#'
-#' # Subset to speed up example
-#' tse <- tse[, colData(tse)$subject %in% c("900", "934", "843", "875")]
-#'
-#' tse2 <- getBaselineDivergence(tse,
-#'                               group = "subject",
-#'                               time_field = "time",
-#'                               name_divergence = "divergence_from_baseline",
-#'                               name_timedifference = "time_from_baseline",
-#'                               assay.type="relabundance",
-#'                               FUN = vegan::vegdist,
-#'                               method="bray")
-#'
-#' tse2 <- getBaselineDivergence(tse,
-#'                               baseline_sample = "Sample-875",
-#'                               group = "subject",
-#'                               time_field = "time",
-#'                               name_divergence = "divergence_from_baseline",
-#'                               name_timedifference = "time_from_baseline",
-#'                               assay.type="relabundance",
-#'                               FUN = vegan::vegdist,
-#'                               method="bray")
-#'
-#' @name getBaselineDivergence
+setGeneric("addBaselineDivergence", signature = "x", function(x, ...)
+    standardGeneric("addBaselineDivergence"))
+
+#' @rdname addBaselineDivergence
 #' @export
-getBaselineDivergence <- function(x,
-                            group=NULL,
-                            time_field,
-                            name_divergence = "divergence_from_baseline",
-                            name_timedifference = "time_from_baseline",			    
-                            assay.type = "counts",
-                            FUN = vegan::vegdist,
-                            method="bray",
-                            baseline_sample=NULL,
-                            altexp = NULL,
-                            dimred = NULL,
-                            n_dimred = NULL,
-                            ...){
+setMethod("addBaselineDivergence", signature = c(x = "SummarizedExperiment"),
+    function(
+        x,
+        time.col,
+        assay.type = "counts",
+        group = NULL,
+        name = "divergence",
+        name.time = "time_diff",
+        method = "bray",
+        dimred = NULL,
+        ndimred = NULL,
+        dis.fun = vegan::vegdist,
+        baseline_sample = NULL,
+        ...){
```

Comment:
This will be taken into account in `mia::getDivergence()`

Reply:
https://github.com/microbiome/mia/pull/653

Reply:
Sry for delays - I dont think it is very common and not necessarily so good practice to calculate divergences from the results of dimension reduction techniques like PCA. It could be sometimes useful to illustrate stuff in PCA space, so if this was already added then it can be there. Otherwise, possible to add that later if needs arise.

Reply:
Yes, it is added in `getDivergence` as "hidden" option